libcontainer/configs/validate/validator.go

```diff
@@ -259,6 +260,39 @@ func cgroupsCheck(config *configs.Config) error {
 	return nil
 }
 
+func mountsStrict(config *configs.Config) error {
+	for _, m := range config.Mounts {
+		if !m.IsIDMapped() {
+			continue
+		}
+
+		if !m.IsBind() {
+			return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is supported only for mounts with the option 'bind'", m)
+		}
+		if config.RootlessEUID {
+			return fmt.Errorf("invalid mount %+v: GidMappings/UidMappings is not supported when runc is being launched with EUID != 0, needs CAP_SYS_ADMIN on the runc parent's user namespace", m)
+		}
+		if config.UidMappings == nil || config.GidMappings == nil {
+			return fmt.Errorf("invalid mount %+v: not yet supported to use GidMappings/UidMappings in a mount without also using a user namespace", m)
+		}
+		if !sameMapping(config.UidMappings, m.UIDMappings) {
+			return fmt.Errorf("invalid mount %+v: not yet supported for the mount uidMappings to be different than user namespace uidMapping", m)
+		}
+		if !sameMapping(config.GidMappings, m.GIDMappings) {
+			return fmt.Errorf("invalid mount %+v: not yet supported for the mount gidMappings to be different than user namespace gidMapping", m)
+		}
```

Comment:
I thought the same. But other errors are already printing the whole struct and my thinking was: most users don't use runc directly, they use it via containerd or kubernetes, and in those cases it is not easy to see what the config.json has for a user. So keeping the complete error message will help us to know what is wrong and debug users bug reports (like using X flag and some other option that is not allowed).

So I thought it was better to keep the whole struct. It should really never happen (so not possible this will be spamming logs nor anything, these are errors that stop the execution) and whenever they do happen, better to have all the info.

What do you think?

Reply:
I take that reaction as an ok, so I kept this